woocommerce/woocommerce
# Addressing invalid import in types when imported from external project.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This is a small change to address an issue preventing the types from working correctly when it's imported outside of the mono-repo. Currently the types as they're generated include a reference to a local `.pnpm` folder, which isn't available outside of the mono-repo:
```js
export const CurrencyContext: import(".pnpm/@types+react@17.0.71/node_modules/@types/react").Context<{
    getCurrencyConfig: () => {
        code: string;
        symbol: string;
        symbolPosition: string;
        decimalSeparator: string;
        priceFormat: string;
        thousandSeparator: string;
        precision: number;
    };
   ...
```

This PR adds the referenced types as a dev dependency to make them available locally when not using pnpm. In my tests just adding this dependency address the issue.

### How to test the changes in this Pull Request:


Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Before checking out this branch, build all on `trunk`.
2. Open this file -> `packages/js/currency/build-types/currency-context.d.ts`
3. Notice that it includes the `import(".pnpm/@types+react@17.0.71/node_modules/@types/react"...` statement.
4. Now checkout this branch, run pnpm install, and delete the `build-types` folder. (`packages/js/currency/build-types`)
5. Rebuild the `@woocommerce/currency` package.
6. Look at the same file above, and you should notice that there is no longer a reference to a `.pnpm` subdirectory. It should now look like this -> `export const CurrencyContext: import("react")` 

### Changelog entry





<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message Adding dependency to avoid invalid types reference in external plugins.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
